src/functions/list-manually-created-documents/handler.ts

```diff
@@ -0,0 +1,46 @@
+import { ValidatedEventAPIGatewayProxyEvent } from "@libs/api-gateway";
+import { middyfy } from "@libs/lambda";
+import { getElastic } from "src/elastic";
+import { timedCurationsService } from "src/database/services";
+import { getElasticCredentialsForSession } from "@libs/auth-utils";
+
+/**
+ * Lambda for listing manually created documents
+ *
+ * @param event event
+ */
+const listManuallyCreatedDocuments: ValidatedEventAPIGatewayProxyEvent<any> = async event => {
+  const { headers: { authorization, Authorization } } = event;
+  const authHeader = Authorization || authorization;
+
+  const auth = await getElasticCredentialsForSession(authHeader);
+  if (!auth) {
+    return {
+      statusCode: 401,
+      body: "Unauthorized"
+    };
+  }
+
+  const elastic = getElastic(auth);
+  if (!(await elastic.hasCurationsAccess())) {
+    return {
+      statusCode: 403,
+      body: "Forbidden"
+    };
+  }
+
+  const manuallyCreatedCurations = await timedCurationsService.listManuallyCreatedDocumentCurations();
+
+  const documents = await Promise.all(
+    manuallyCreatedCurations.map(async curation => {
```

Comment:
This should be done in one request, where you list documents with a list of document IDs as a filter.

Reply:
Done